react-components/src/apps/admin/AdminAppsGridListing.js

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from "react";
+import getAppsSearchRegex from "../appSearchRegex";
+import {
+    AppName,
+    AppStatusIcon,
+    build,
+    DETableRow,
+    EmptyTable,
+    EnhancedTableHead,
+    getMessage,
+    Highlighter,
+    LoadingMask,
+    palette,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+
+import {
+    IconButton,
+    makeStyles,
+    Table,
+    TableBody,
+    TableCell,
+    Typography,
+} from "@material-ui/core";
+import InfoIcon from "@material-ui/icons/InfoOutlined";
+import AppFields from "../AppFields";
+import ids from "../listing/ids";
+import { injectIntl } from "react-intl";
+import messages from "../messages";
+
+const useStyles = makeStyles((theme) => ({
+    toolbarItemColor: {
+        color: palette.darkBlue,
+    },
+    listingFont: {
+        fontSize: 12,
+    },
+    container: {
+        width: "100%",
+        height: "100%",
+        marginTop: 0,
+        overflow: "auto",
+        backgroundColor: palette.white,
+    },
+    heading: {
+        backgroundColor: palette.lightGray,
+        width: "100%",
+        height: 30,
+    },
+}));
+const tableColumns = [
+    {
+        name: AppFields.NAME.fieldName,
+        enableSorting: true,
+        key: AppFields.NAME.key,
+    },
+    {
+        name: AppFields.INTEGRATOR.fieldName,
+        enableSorting: true,
+        key: AppFields.INTEGRATOR.key,
+    },
+    {
+        name: "",
+        enableSorting: false,
+        key: "AppInfo",
+    },
+    {
+        name: "",
+        enableSorting: false,
+        key: "Status",
+    },
+];
+
+function AdminAppGridListing(props) {
+    const {
+        baseId,
+        apps,
+        heading,
+        presenter,
+        searchText,
+        loading,
+        intl,
+    } = props;
+    const [selectedApps, setSelectedApps] = useState([]);
+    const searchRegex = getAppsSearchRegex(searchText);
+    const classes = useStyles();
+    const onRequestSort = () => {
+        console.log("Sorting...");
```

Comment:
Should this be calling back to the presenter so it can sort with the listing service?

Reply:
Lol. I totally forgot to implement sorting. Good 👀